File: programmingModels/OpenMP/fortran/02_target_teams_parallel.f90
program  main
  use omp_lib
  implicit none
  integer i
!$omp target teams distribute parallel do thread_limit(2) num_teams(2)
    do i=1,8
       write(*,*) "Thread", omp_get_thread_num(), &
            "out of", omp_get_num_threads() ,&
            "threads in team", omp_get_team_num(), &
            "out of", omp_get_num_teams(), &
            "teams is using index" , i
     end do
!$omp end target teams distribute parallel do
end program main

File: programmingModels/OpenMP/c/02_target_teams_parallel.cpp
#include <stdio.h>
#include <omp.h>

int main( int argv, char** argc ) {
#pragma omp target teams distribute parallel for simd
  for(int i=0;i<10000;i++)
    {
      printf( "Thread %d out of %d threads in team %d out of %d teams is using index %d\n", 
	      omp_get_thread_num(), omp_get_num_threads(), 
	      omp_get_team_num(), omp_get_num_teams(), i );
    }

return 0;
}
